api/room_resource/schema.py

```diff
@@ -75,6 +75,47 @@ def mutate(self, info, **kwargs):
         return AssignResource(room_resource=room_resources)
 
 
+class UpdateAssignedResource(graphene.Mutation):
+    """
+        Update assigned resource in a room
+    """
+    class Arguments:
+        resource_id = graphene.Int(required=True)
+        room_id = graphene.Int(required=True)
+        quantity = graphene.Int(required=True)
+    room_resource = graphene.Field(RoomResource)
+
+    @Auth.user_roles('Admin')
+    def mutate(self, info, resource_id, room_id, **kwargs):
+        room_resource_query = RoomResource.get_query(info)
+        room = room_resource_query.filter(
+            RoomResourceModel.room_id == room_id).all()
+        if not room:
+            raise GraphQLError('Room has no assigned resource')
+        resource = [resource for resource in room if resource.resource_id == resource_id]  # noqa
```

Comment:
For readability purposes, perhaps this could be a normal nested for loop?

Reply:
Thanks, @d-kahara 
This has been worked on